lib/jxl/enc_ans.cc

```diff
@@ -115,24 +115,26 @@ const uint8_t kLogCountSymbols[ANS_LOG_TAB_SIZE + 2] = {
 
 // Returns the difference between largest count that can be represented and is
 // smaller than "count" and smallest representable count larger than "count".
-int SmallestIncrement(uint32_t count, uint32_t shift) {
-  int bits = count == 0 ? -1 : FloorLog2Nonzero(count);
-  int drop_bits = bits - GetPopulationCountPrecision(bits, shift);
-  return drop_bits < 0 ? 1 : (1 << drop_bits);
+// Should be invoked with `count > 0` only
```

Comment:
Better to make that a `JXL_DASSERT()` instead of a comment. Or rename the function to `SmallestIncrementNonzero` like `FloorLog2Nonzero`.

Reply:
Done both. BTW, do you know the reasoning for overestinmation of total_pixels value (https://github.com/libjxl/libjxl/pull/4135)?